### PR TITLE
feat: add cli config manager

### PR DESCRIPTION
**Pull request by ravisuhag**
- Add config init and list command. 
- Load CLI config in-home path or with the path specified via env variable. 
- Display message if config is not intialised.

### CODE REVIEW

File: cmd/config.go [hunk 1]
```diff
@@ -2,43 +2,96 @@ package cmd
...
 func configCommand() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "config",
-		Short: "manage guardian CLI configuration",
+		Use:   "config <command>",
+		Short: "Manage client configuration settings",
+		Example: heredoc.Doc(`
+			$ stencil config init
+			$ stencil config get
+			$ stencil config list`),
```
Comment (rahmatrhd) [inline cmd/config.go:19]: need to update the examples

File: cmd/config.go [hunk 2]
```diff
@@ -2,43 +2,96 @@ package cmd
...
 	return cmd
 }
 
 func configInitCommand() *cobra.Command {
 	return &cobra.Command{
 		Use:   "init",
-		Short: "initialize CLI configuration",
+		Short: "Initialize a new client configuration",
+		Example: heredoc.Doc(`
+			$ stencil init --path=.stencil.yml
```
Comment (rahmatrhd) [inline cmd/config.go:34]: update example

File: cmd/config.go [hunk 3]
```diff
@@ -2,43 +2,96 @@ package cmd
...
+		},
+	}
+}
+
+func configGetCommand() *cobra.Command {
+	var cmd = &cobra.Command{
+		Use:   "get <key>",
+		Short: "Update configuration with a value for the given key",
+		Example: heredoc.Doc(`
+			$ stencil config get --path=.stencil.yml
```
Comment (rahmatrhd) [inline cmd/config.go:57]: update example

File: cmd/config.go [hunk 4]
```diff
@@ -2,43 +2,96 @@ package cmd
...
+}
 
-			filepath := fmt.Sprintf("%v", app.CLIConfigFile)
-			if err := ioutil.WriteFile(filepath, b, 0655); err != nil {
+func configListCommand() *cobra.Command {
+	var cmd = &cobra.Command{
+		Use:   "list",
+		Short: "Update configuration with a value for the given key",
+		Example: heredoc.Doc(`
+			$ stencil config list --path=.stencil.yml
```
Comment (rahmatrhd) [inline cmd/config.go:79]: update example

File: cmd/config.go [hunk 5]
```diff
@@ -2,43 +2,96 @@ package cmd
...
+		`),
+		Annotations: map[string]string{
+			"group:core": "true",
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			config, err := app.LoadCLIConfig()
 			if err != nil {
 				return err
 			}
+			fmt.Println(config)
```
Comment (rahmatrhd) [inline cmd/config.go:67]: this get command isn't returning one specified config key

**Review by rahmatrhd — CHANGES_REQUESTED** [review]

File: cmd/config.go (hunk 1 above)
Comment (ravisuhag) [reply cmd/config.go]: Updated.

File: cmd/config.go (hunk 3 above)
Comment (ravisuhag) [reply cmd/config.go]: Updated.

File: cmd/config.go (hunk 5 above)
Comment (ravisuhag) [reply cmd/config.go]: Removed get command for now.

File: cmd/config.go (hunk 4 above)
Comment (ravisuhag) [reply cmd/config.go]: Updated.

**Comment by ravisuhag** [comment]
@rahmatrhd Fixed the examples. Please check.

**Review by rahmatrhd — APPROVED** [review]
LGTM

**Pull request merged**